### PR TITLE
Replace use of np.allclose in tests.

### LINKED ISSUE #34096: Adding and use `assertAllClose` API in our tests.
Currently we have many uses of 
```
self.assertEqual(ans.shape, expected.shape)
self.assertTrue(np.allclose(ans.cpu(), expected, rtol=1e-02, atol=1e-03, equal_nan=True))
``` 
in our tests. e.g. https://github.com/pytorch/pytorch/blob/master/test/test_torch.py#L11270
`torch.allclose` won't work well here as it only accept two tensors.
But this way is hard for backends like XLA to override precisions :P 
We want to propose make an API
```
self.assertAllClose(torch_tensor, np_array, rtol, atol, equal_nan)
```
to replace it. 
@mruberry @gchanan @ezyang Does this sound good? 
cc: @JackCaoG

### PR DESCRIPTION
**Pull request by ailzhang**
fixes #34096

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -906,6 +906,9 @@ def assertTensorsEqual(a, b):
         elif isinstance(x, bool) and isinstance(y, bool):
             super(TestCase, self).assertEqual(x, y, message)
         elif isinstance(x, Number) and isinstance(y, Number):
+            # Handle Nans
+            if x != x and y != y:
+                return
```
Comment (mmphego) [inline torch/testing/_internal/common_utils.py]: What is the likeliness of this code ever returning?

Comment (ailzhang) [reply torch/testing/_internal/common_utils.py]: @mmphego It returns when both `x` and `y` are Nans.

**Comment by ailzhang** [comment]
I'll put this PR on hold until changes in https://github.com/pytorch/pytorch/pull/34258 is merged.

**Review by ezyang — APPROVED** [review]

File: torch/testing/_internal/common_utils.py (hunk 1 above)
Comment (mmphego) [reply torch/testing/_internal/common_utils.py]: Thank you for the feedback.

**Review by mmphego — APPROVED** [review]

**Comment by ailzhang** [comment]
Since #34258 is merged, I've rebase this PR on top of master and landing soon...

**Pull request closed without merging**